src/components/ha-date-range-picker.ts

```diff
@@ -133,41 +149,61 @@ export class HaDateRangePicker extends LitElement {
       <date-range-picker
         ?disabled=${this.disabled}
         ?auto-apply=${this.autoApply}
-        ?time-picker=${this.timePicker}
+        time-picker=${this.timePicker}
         twentyfour-hours=${this._hour24format}
         start-date=${this.startDate}
         end-date=${this.endDate}
         ?ranges=${this.ranges !== false}
+        opening-direction=${this._openingDirection}
         first-day=${firstWeekdayIndex(this.hass.locale)}
       >
         <div slot="input" class="date-range-inputs">
-          <ha-svg-icon .path=${mdiCalendar}></ha-svg-icon>
-          <ha-textfield
-            .value=${this.timePicker
-              ? formatDateTime(
-                  this.startDate,
-                  this.hass.locale,
-                  this.hass.config
-                )
-              : formatDate(this.startDate, this.hass.locale, this.hass.config)}
-            .label=${this.hass.localize(
-              "ui.components.date-range-picker.start_date"
-            )}
-            .disabled=${this.disabled}
-            @click=${this._handleInputClick}
-            readonly
-          ></ha-textfield>
-          <ha-textfield
-            .value=${this.timePicker
-              ? formatDateTime(this.endDate, this.hass.locale, this.hass.config)
-              : formatDate(this.endDate, this.hass.locale, this.hass.config)}
-            .label=${this.hass.localize(
-              "ui.components.date-range-picker.end_date"
-            )}
-            .disabled=${this.disabled}
-            @click=${this._handleInputClick}
-            readonly
-          ></ha-textfield>
+          ${!this.minimal
```

Comment:
This already is a slot, so you can pass whatever element you want in here.

Reply:
I don't understand what the problem is. Is this about the order in which the condition and the slot-div are nested? From a readability standpoint I would prefer the current solution. Does this repo have a standard for this?

Reply:
I would not put this in the `ha-date-range-picker` element, but use the slot to override the default (or maybe we should not provide a default at all and let the consumer also use the slot.

So you would get:
```html
<ha-date-range-picker>
  <ha-icon-button slot="trigger"></ha-icon-button>
</ha-date-range-picker>
```

And for the other usecase:
```html
<ha-date-range-picker>
  <ha-textfield readonly slot="trigger" ...></ha-textfield>
  <ha-textfield readonly slot="trigger" ...></ha-textfield>
</ha-date-range-picker>
```
```

Reply:
The slot does not appear to be available within instances of `ha-date-range-picker`. As far as I can tell (I'm not that familiar with Lit), it can be made available by forwarding it using another slot like this (with default option):

````html
<slot name="input/trigger" slot="input">
  <div class="date-range-inputs">
    [.... default icon + 2 textfield option ...]         
  </div>
</slot>
````

Removing the default option should be its own PR. I'm trying to think of use cases in which a custom trigger(slotted element) would be usefull but can't come up with any. Restricting this functionality may be good for consistency. The second option you've provided will also introduce code clones, which may be undesirable.

Is the above-provided solution fine or should we just leave the current solution as is?
(is there a better way of doing this, that I'm not aware of?)